test/integration/driver.go

```diff
@@ -232,28 +231,34 @@ func getRestartEpoch(pod string) (int, error) {
 	return 0, fmt.Errorf("could not obtain envoy restart epoch")
 }
 
-func addConfig(config []byte, kind, name string) {
+func addConfig(config []byte, kind, name string, create bool) {
 	log.Println("Add config")
 	log.Println(string(config))
-	out, err := yaml.YAMLToJSON(config)
-	check(err)
 	istioKind, ok := model.IstioConfig[kind]
 	if !ok {
 		check(fmt.Errorf("Invalid kind %s", kind))
 	}
-	v, err := istioKind.FromJSON(string(out))
+	v, err := istioKind.FromYAML(string(config))
 	check(err)
-	check(istioClient.Put(model.Key{
+	key := model.Key{
 		Kind:      kind,
 		Name:      name,
 		Namespace: params.namespace,
-	}, v))
+	}
+	if create {
+		check(istioClient.Post(key, v))
+	} else {
+		check(istioClient.Put(key, v))
+	}
 }
```

Comment:
Yes, we need PATCH semantics to make "apply" work, and there the difficulty is dealing with lists in protos. That's the path kubectl went on, when a similar question was brought up (why have "create" and "replace"). It's also nice to keep each command to one atomic operation.